tests/system/src/test/java/test/com/sun/javafx/tk/quantum/SystemMenuBarTest.java

```diff
@@ -172,4 +179,77 @@ private void createMenuBarWithItemsStage() {
         t.start();
     }
 
+    @Test
+    public void testFocusMemoryLeak() throws InterruptedException {
+        Util.runAndWait(() -> {
+            Thread.currentThread().setUncaughtExceptionHandler((t,e) -> {
+                e.printStackTrace();
+                failed.set(true);
+                memoryFocusLatch.countDown();
+            });
+            createAndRefocusMenuBarStage();
+        });
+        memoryFocusLatch.await();
+        assertFalse(failed.get());
+    }
+
+    public void createAndRefocusMenuBarStage() {
+        Stage stage = new Stage();
+        VBox root = new VBox();
+
+        final MenuBar menuBar = new MenuBar();
+        final Menu menu = new Menu("MyMenu");
+        menuBar.getMenus().add(menu);
+        menuBar.setUseSystemMenuBar(true);
+        root.getChildren().add(menuBar);
+
+        Scene scene = new Scene(root);
+        stage.setScene(scene);
+        stage.show();
+        final ArrayList<WeakReference<MenuBase>> uncollectedMenus = new ArrayList<>();
+        GlassSystemMenuShim gsmh = new GlassSystemMenuShim();
+        Menu m1 = new Menu("Menu");
+
+        MenuBase menuBase = GlobalMenuAdapter.adapt(m1);
+        ArrayList<MenuBase> menus = new ArrayList<>();
+        gsmh.createMenuBar();
+        for (int i = 0; i < 100; i++) {
+            Platform.runLater(() -> {
+                gsmh.setMenus(List.of(menuBase));
+            });
+        }
+        Platform.runLater(() -> {
+            int strongCount = 0;
+            final List<WeakReference<com.sun.glass.ui.Menu>> u2 = gsmh.getWeakMenuReferences();
+            for (WeakReference<com.sun.glass.ui.Menu> wr : u2) {
+                if (!JMemoryBuddy.checkCollectable(wr)) {
+                    strongCount++;
+                    assertTrue("Too much refs", strongCount < 2);
+                }
+            }
+            assertEquals(1, strongCount, "Exactly one reference should be reachable");
+            memoryFocusLatch.countDown();
+        });
+    }
+
+    public MenuBar createSimpleMenuBar() {
+        MenuBar menuBar = new MenuBar();
+
+        menuBar.setUseSystemMenuBar(true);
+
+        Menu systemMenu = new Menu("systemMenu");
+        menuBar.getMenus().add(systemMenu);
+
+        var newItem = new MenuItem();
+        newItem.setVisible(false);
+        systemMenu.getItems().add(newItem);
+
+        Platform.runLater(() -> {
+            javafx.scene.control.Menu systemMenuContributions = new Menu("123");
+            systemMenu.getItems().add(systemMenuContributions);
+        });
+
+        return menuBar;
+    }
+
```

Comment:
extra newline?

Reply:
Not sure I understand where you want to add or remove a newline?

Reply:
please remove extra newline.